packages/material-ui/test/typescript/components.spec.tsx

```diff
@@ -163,7 +163,7 @@ const BottomNavigationTest = () => {
   const value = 123;
 
   return (
-    <BottomNavigation value={value} onChange={e => log(e)} showLabels>
```

Comment:
The purpose of writing these as lambdas was to ensure that the argument type was being inferred...

Reply:
Shouldn't it complain that `log` is not assignable to `(e: EventType) => ReturnType` in those cases? What makes this different from the cases where you added an explicit `ExpectType`?

Reply:
>Shouldn't it complain that `log` is not assignable to `(e: EventType) => ReturnType` in those cases?

Perhaps it should, but it doesn't.

>What makes this different from the cases where you added an explicit `ExpectType`?

It's a less rigorous test (just that something other than `any` was inferred for the argument, not making an assertion about what _was_ inferred). There's no reason an `ExpectType` couldn't be added also.

Reply:
Could you quickly push a test that is not throwing when it should for my own peace of mind? Just want to make sure we're not disabling some lint rule that was removed because of something unrelated.

Reply:
Hm, I don't think I follow. Could you elaborate?

Reply:
Going back to your comment:

> Shouldn't it complain that `log` is not assignable to `(e: EventType) => ReturnType` in those cases?

I actually don't understand this... in what case should that error occur and why?

Reply:
> The purpose of writing these as lambdas was to ensure that the argument type was being inferred...

This could've been done by passing a method that exactly matches to `onChange` signature. I would prefer this over disabling a lint rule globally.

Reply:
That doesn’t test inference because you’re passing a function whose signature is already determined. The point is to pass an inline function with no parameter signature so that if it can’t infer the type there will be a `no-implicit-any` error.

Reply:
> The point is to pass an inline function with no parameter signature so that if it can’t infer the type there will be a `no-implicit-any` error.

Ah now I understand. Happy to revert a7f6d4c806f41c576b3a52f728d6a3c0ccb03082 then.

Reply:
If we didn’t want to disable the lint rule we could just as well pass `e => {}` to test this, the `log` call is unimportant But IIRC the only reason for it was to appease another lint rule for unused variables.

Probably should just selectively disable the lint rule for these lines.

Reply:
I'm ok with disabling the rule since our ts files never make it to runtime anyway. I would say that the only useful pattern the lint rule flags is in react props. A new callback could defeat memoized components. 

The ts files in our docs are transpiled to JS at which point eslint would cover that case.

If end up disabling the rules for every instance of unnecessary callbacks then we might as well disable it globally (or maybe starting with per file disables)

Reply:
Ok, I reverted 79c4872.